### PR TITLE
fix: user roles take single string & DDL readPreference tests (master)

### PR DESCRIPTION
**Pull request by reggi**
Bringing in changes from https://github.com/mongodb/node-mongodb-native/pull/2547

### CODE REVIEW

File: test/functional/readpreference.test.js [hunk 1]
```diff
@@ -250,7 +263,9 @@ describe('ReadPreference', function () {
 
     test: function (done) {
       var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
+      var client = configuration.newClient(configuration.writeConcernMax(), {
+        poolSize: 1
+      });
```
Comment (nbbeeken) [inline test/functional/readpreference.test.js:268]: Why are these changing if the code is the same? Maybe double check prettier version?

**Review by nbbeeken — CHANGES_REQUESTED** [review]
Looks good, consistent with the other PR, except some formatting that I think is a mistake? lmk

File: test/functional/readpreference.test.js (hunk 1 above)
Comment (reggi) [reply test/functional/readpreference.test.js]: fixed!

File: test/functional/readpreference.test.js [hunk 2]
```diff
@@ -587,4 +588,144 @@ describe('ReadPreference', function () {
...
+              done();
+            };
+            parent[method].apply(parent, [...args, callback]);
+          });
+        }
+      });
+    });
+  });
+
+  context('should enforce fixed primary read preference', function () {
```
Comment (mbroadst) [inline test/functional/readpreference.test.js:662]: it looks like you have two copies of the same test additions

**Review by mbroadst — CHANGES_REQUESTED** [review]

File: test/functional/readpreference.test.js (hunk 2 above)
Comment (reggi) [reply test/functional/readpreference.test.js:662]: doh, I was wrestling with this file / linting and manually doubled up.

**Review by mbroadst — APPROVED** [review]
LGTM

**Review by nbbeeken — APPROVED** [review]
LGTM

**Review by emadum — APPROVED** [review]

**Pull request merged**